### PR TITLE
change prefix in piler.js

### PR DESCRIPTION
**Pull request by th-2021**
This patch sets base_url in piler.js according to the setting on config-site.php.

### CODE REVIEW

**Review by jsuto — CHANGES_REQUESTED** [review]
I'd rather like to see a simpler solution. Eg. just set PATH_PREFIX as a docker-compose env var, and if it exists in the container then you simply use sed to fix piler.js, and update config-site.php with this value, eg. ```$config['PATH_PREFIX'] = ....;``` if it's not already in the file.

Also, please use $( ... ) format for running commands, not the obsoleted backtick (`)

**Comment by th-2021** [comment]
what about $config['SITE_URL'] = 'http://' . $config[SITE_NAME_CONST] . '/mail/';
Should I change util/config-site.php.in to have PATH_PREFIX in it and replace it in start with the real PATH_PREFIX?

**Comment by jsuto** [comment]
Yes, you are correct. I suggest to add ```config['PATH_PREFIX'] = '/';``` to util/config-site.php.in before the SITE_URL line.

**Comment by jsuto** [comment]
And also fix ```etc/config-site.dist.php``` as well, please.

File: docker/start.sh [hunk 1]
```diff
@@ -159,6 +159,17 @@ fix_configs() {
    if ! grep "'SPHINX_HOSTNAME_READONLY'" "$CONFIG_SITE_PHP"; then
       echo "\$config['SPHINX_HOSTNAME_READONLY'] = '${MANTICORE_HOSTNAME}:9307';" >> "$CONFIG_SITE_PHP"
    fi
+
+   # Fix for PATH_PREFIX
+   if [[ -v "$PATH_PREFIX" ]];
```
Comment (jsuto) [inline docker/start.sh:164]: The proper syntax is the following.

```if [[ -v PATH_PREFIX ]]; then```

File: docker/start.sh [hunk 2]
```diff
@@ -159,6 +159,17 @@ fix_configs() {
...
+     then
+       log "PATH_PREFIX set $PATH_PREFIX"
+       sed -i -e "s#location.origin\ +\ .*#location.origin\ +\ $PATH_PREFIX,#" /var/piler/www/assets/js/piler.js
+       sed -i "s%PATH_PREFIX%${PATH_PREFIX}%" "$CONFIG_SITE_PHP"
```
Comment (jsuto) [inline docker/start.sh:168]: This won't cut it, see my next comment below.

File: etc/config-site.dist.php [hunk 3]
```diff
@@ -3,7 +3,7 @@
 define('SITE_NAME_CONST', 'SITE_NAME');
 
 $config[SITE_NAME_CONST] = 'HOSTNAME';
-$config['SITE_URL'] = 'http://' . $config[SITE_NAME_CONST] . '/';
+$config['SITE_URL'] = 'http://' . $config[SITE_NAME_CONST] . 'PATH_PREFIX';
```
Comment (jsuto) [inline etc/config-site.dist.php:6]: PATH_PREFIX is a non-existing variable. What we need is

```
$config['PATH_PREFIX'] = '/';
$config['SITE_URL'] = 'http://' . $config[SITE_NAME_CONST] . $config['PATH_PREFIX'];
```

Then in start.sh you need to fix ```$config['PATH_PREFIX'] = '/';```

**Review by jsuto — CHANGES_REQUESTED** [review]

File: etc/config-site.dist.php (hunk 3 above)
Comment (th-2021) [reply etc/config-site.dist.php:6]: I used PATH_PREFIX like HOSTNAME in the file. and replaced it with the env variable in start.sh I also added $config['PATH_PREFIX'].
In my setup it looks like:
...
$config['SITE_URL'] = 'http://' . $config[SITE_NAME_CONST] . '/mail/';
...
$config['PATH_PREFIX'] = '/mail/';
...

Adding $config['PATH_PREFIX'] is an little bit more tricky.

**Comment by jsuto** [comment]
OK, it looks good. However, there's an issue, see #167. The problem is that with the current setup you can't override the BRANDING_* variables, so they need to be restored to the original place as they were. Please check out the other PR, and we need to come to an agreement how to solve this catch 22 issue.

**Pull request merged**